### DISCUSSION TITLE
fail2ban make decrease web-server RPS

### DISCUSSION THREAD

**Discussion by antoshkin** [fail2ban/fail2ban#3668] [General]
If I turn _off_ fail2ban, then I get ~_400_ requests per second. (tested by **ab** utility).

If I turn _on_ fail2ban, then I get ~_300_ requests per second.

How can I fix it?

**Comment by sebres** [comment]
Fix what exactly?

Every host has a restricted resources as for CPU, memory (cache) etc. The extra load affected by fail2ban depends on many factors, inclusive quality of regexs and dealing with "parasitic" log-trafic (in sense of fail2ban matches).
Take a look at [wiki :: Best practice](https://github.com/fail2ban/fail2ban/wiki/Best-practice) for details.

Particularly the quality of REs is extremely important - for instance weak, unanchored REs with many catch-alls may be very CPU-hungry.

However if one thinks that the security is for free - it is wrong assumption (for example take a look at PC with and without antivirus).

But additionally to security, an important question would be also - if IDS/IPS would not ban evildoers, how it would affect the load average over the long term - more or less it is to compare the load generated by not banned evildoers (the costs of flood, brute force etc) with the load occurred through IDS/IPS.

Anyway it is impossible to help you here without details (used filters/REs, log-excerpts, etc).
